site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -0,0 +1,184 @@
+```
+
+Before we write the query, let's investigate each column in the `instance` table. As you've already noticed, we create AWS resources by `INSERT`ing rows into corresponding tables. For example, we created a security group by executing the `INESRT INTO security_group` query. With that in mind, let's look into the columns of the `instance` table (which is in charge of managing EC2 servers) that will be used in our query:
+- `ami`: this column is the [Amazon Machine Image](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) ID that is used to launch the EC2 server. We can either directly insert the AMI ID (like `ami-0f1a5f5ada0e7da53`), or use a parameter reference to the AWS Systems Manager Parameter Store (like `resolve:ssm:/aws/service/canonical/ubuntu/server/22.04/stable/current/amd64/hvm/ebs-gp2/ami-id`). We'll go with the parameter reference choice.
+
+:::info
+The parameter reference `resolve:ssm:/aws/service/canonical/ubuntu/server/22.04/stable/current/amd64/hvm/ebs-gp2/ami-id` uses the `resolve` syntax to retrieve the value of an [SSM](https://docs.aws.amazon.com/systems-manager/latest/userguide/what-is-systems-manager.html) parameter.
+
+The `ssm` protocol specifies that the parameter is stored in the AWS Systems Manager Parameter Store. The rest of the path `/aws/service/canonical/ubuntu/server/22.04/stable/current/amd64/hvm/ebs-gp2/ami-id` is the path to the SSM parameter containing the Amazon Machine Image (AMI) ID for a specific Ubuntu Server 22.04 AMI that is optimized for use with Amazon Elastic Block Store (EBS) General Purpose SSD volumes.
+:::
+
+:::info
+AWS Resolve syntax is a feature that allows you to retrieve and use the values of AWS Systems Manager Parameter Store parameters, AWS Secrets Manager secrets, and Amazon S3 bucket objects in AWS SDK and Amazon EC2 user data scripts.
+
+The AWS Resolve syntax uses the following format:
+```
+resolve:<protocol>:<value>
+```
+- `protocol`: Specifies the type of resource that is being referenced. The currently supported protocols are `ssm` for AWS Systems Manager Parameter Store, `secretsmanager` for AWS Secrets Manager secrets, and `s3` for Amazon S3 bucket objects.
+- `value`: Specifies the path to the resource that is being referenced.
+
+Using the AWS Resolve syntax in AWS SDK or EC2 user data script allows you to dynamically reference and use the values of resources in other AWS services, which can be useful for automating the deployment of AWS resources.
+
+For example, you could use `resolve:ssm:/my/parameter/path` to reference a parameter stored in AWS Systems Manager Parameter Store, `resolve:secretsmanager:/my/secret/path` to reference a secret stored in AWS Secrets Manager, or `resolve:s3:/my/bucket/path/myobject.txt` to reference an object stored in an Amazon S3 bucket.
+:::
+
+- `instance_type`: this column contains the [instance type](https://aws.amazon.com/ec2/instance-types/) of the EC2 server. We'll use a `t2.small` instance for our deployment.
```

Comment:
does not work on a `t2.micro`, right? just to see if we can stick to the free tier.

Reply:
I'll investigate this. Makes sense to stick to free tier.